README.md

```diff
@@ -1,80 +1,203 @@
 
-###Debugging Unexpected Failures
+#### Create a new app from an image (method 3) 
 
-Review some of the common tips and suggestions [here](https://github.com/openshift/origin/blob/master/docs/debugging-openshift.md).
+You may have noticed the index page "Page view count" reads "No database configured". Let's fix that by adding a MongoDB service. We could use the second OpenShift template example (`nodejs-mongodb.json`) but for the sake of demonstration let's point `oc new-app` at a DockerHub image:
+
+        $ oc new-app centos/mongodb-26-centos7 \
+        $ -e MONGODB_USER=admin,MONGODB_DATABASE=mongo_db,MONGODB_PASSWORD=secret,MONGODB_ADMIN_PASSWORD=super-secret
+        
+The `-e` flag sets the environment variables we want used in the configuration of our new app.
+
+Running `oc status` or checking the web console will reveal the address of the newly created MongoDB:
+
+	In project nodejs-echo on server https://10.2.2.2:8443
+	
+	svc/mongodb-26-centos7 - 172.30.0.112:27017
+	  dc/mongodb-26-centos7 deploys istag/mongodb-26-centos7:latest
+	    deployment #1 running for 43 seconds - 1 pod
+	
+	http://10.2.2.2 to pod port 8080-tcp (svc/nodejs-ex)
+	  dc/nodejs-ex deploys istag/nodejs-ex:latest <-
+	    bc/nodejs-ex builds https://github.com/openshift/nodejs-ex with openshift/nodejs:0.10
+	    deployment #1 deployed 14 minutes ago - 1 pod
+            
+Note that the url for our new Mongo instance, for our example, is `172.30.0.112:27017`, your's will likely differ.
+
+#### Setting environment variables
+
+To take a look at environment variables set for each pod, run `oc env pods --all --list`.
+
+We need to add the environment variable `MONGO_URL` to our Node.js web app so that it will utilize our MongoDB, and enable the "Page view count" feature. Run:
 
+        $ oc env dc/nodejs-ex MONGO_URL='mongodb://admin:secret@172.30.0.112:27017/mongo_db'
+        
+Then check `oc status` to see that an updated deployment has been kicked off:
+        
+	In project nodejs-echo on server https://10.2.2.2:8443
+	
+	svc/mongodb-26-centos7 - 172.30.0.112:27017
+	  dc/mongodb-26-centos7 deploys istag/mongodb-26-centos7:latest
+	    deployment #1 deployed 2 hours ago - 1 pod
+	
+	http://10.2.2.2 to pod port 8080-tcp (svc/nodejs-ex)
+	  dc/nodejs-ex deploys istag/nodejs-ex:latest <-
+	    bc/nodejs-ex builds https://github.com/openshift/nodejs-ex with openshift/nodejs:0.10
+	    deployment #2 deployed about a minute ago - 1 pod
+	    deployment #1 deployed 2 hours ago
+
+#### Success
 
-### Web UI ###
+You should now have a Node.js welcome page showing the current hit count, as stored in a MongoDB database.
+
+#### Pushing updates
+
+Assuming you used the URL of your own forked report, we can easily push changes to that hosted repo and simply repeat the steps above to build which will trigger the new built image to be deployed.
```

Comment:
sorry, i'm using regex syntax.  "report" should be "repository" and "new" should be "newly" in the sentence above.